spec/formalism/form/coercion_spec.rb

```diff
@@ -0,0 +1,199 @@
+# frozen_string_literal: true
+
+## This is unit test, created to simplify testing of different coercions
+## You can find integration test in spec/formalism/form/fields_spec.rb
+## (Formalism::Form -> .field -> coersion)
+##
+## In tests `eql` used instead of `eq` to determine result type
+describe 'Formalism::Coercion' do
+	let(:value) {}
+
+	let(:field_defenition) { [type] }
+	let(:form) do
+		## Pass method call result throug instance_eval inside Class.new
+		field_defenition = self.field_defenition
+
+		Class.new(Formalism::Form) do
+			field :value, *field_defenition
+			alias_method :execute, :value
+		end
+	end
+
+	subject(:coercion_result) { form.run(value: value).result }
+
+	shared_examples 'it parses nil' do |expected_result = nil|
+		let(:value) { nil }
+		it { is_expected.to eql(expected_result) }
+	end
+
+	shared_examples 'it parses empty string' do |expected_result = nil|
+		let(:value) { '' }
+		it { is_expected.to eql(expected_result) }
+	end
+
+	shared_examples 'it parses empty array' do |expected_result = nil|
+		let(:value) { [] }
+		it { is_expected.to eql(expected_result) }
+	end
+
+	shared_examples 'it parses empty hash' do |expected_result = nil|
+		let(:value) { {} }
+		it { is_expected.to eql(expected_result) }
```

Comment:
Please, separate `let` with other things like `it` with empty line, like you do so for multiple `it`s.

Reply:
created #5 to prevent this in future